desc/magnetic_fields/_current_potential.py

```diff
@@ -842,3 +1042,790 @@ def nfp_loop(j, f):
+    npts : int
+        number of points to discretize the current potential with in the
+        zeta direction. The discretization in theta will be proportional to this.
+        The discretized current potential is then passed to the
+        ``skimage.measure.find_contours`` which finds contours using a
+        marching squares algorithm.
+    show_plots : bool, optional
+        whether to plot the contours, useful for debuggin or seeing why contour finding
+        fails, by default False
+    stell_sym : bool, optional
+        if the modular coilset has stellarator symmetry, by default False.
+        Does nothing for helical coils.
+
+    Returns
+    -------
+    contour_theta, contour_zeta: list of 1D arrays
+        list of length num_coils containing arrays of the theta
+        and zeta values describing each contour found.
+    """
+    ################################################################
+    # find current helicity
+    ################################################################
+    # we know that I = -(G - G_ext) / (helicity * NFP)
+    # if net_toroidal_current is zero, then we have modular coils,
+    # and just make helicity zero
+    net_poloidal_current = surface_current_field.G
+    net_toroidal_current = surface_current_field.I
+    nfp = surface_current_field.NFP
+    helicity = safediv(net_poloidal_current, net_toroidal_current * nfp, threshold=1e-8)
+    dz = 2 * np.pi / nfp / npts
+    if not jnp.isclose(helicity, 0):
+        # helical coils
+        zeta_full = jnp.arange(
+            0,
+            2 * jnp.pi / nfp + 1e-6,
+            dz,
+        )
+        # ensure we have always have points at least from -2pi, 2pi as depending
+        # on sign of I, the contours from Phi = [0, abs(I)] may have their starting
+        # points (the theta value at zeta=0) be positive or negative theta values,
+        # and we want to ensure we catch the start and end of the contours
+        theta_full = jnp.arange(
+            jnp.sign(helicity) * 2 * jnp.pi,
+            -jnp.sign(helicity) * (2 * np.pi * int(np.abs(helicity) + 1) + 1e-6),
+            -jnp.sign(helicity) * 2 * np.pi / npts / nfp,
+        )
+
+        theta_full = jnp.sort(theta_full)
+    else:
+        # modular coils
+        theta_full = jnp.linspace(0, 2 * jnp.pi, npts + 1)
+        # we start below 0 for zeta to allow for contours which may go in/out of
+        # the zeta=0 plane
+        zeta_full = jnp.arange(-jnp.pi / nfp, (2 + 1) * jnp.pi / nfp, dz)
+
+    ################################################################
+    # find contours of constant phi
+    ################################################################
+    # make linspace contours
+    if not jnp.isclose(helicity, 0):
+        # helical coils
+        # we start them on zeta=0 plane, so we will find contours
+        # going from 0 to I (corresponding to zeta=0, and theta*sign(I) increasing)
+        contours = jnp.linspace(
```

Comment:
linearly spaced from 0 to I gives you contours which have beginning points at zeta=0 which cover the full poloidal extent, if you have helical coils. 

Modular, going from 0 -> G / nfp (from below) yields the same thing but spaced out in zeta over one field period (or over one half field period if stellarator symmetric).

even spacing in Phi is a logical way to choose which coils to make from the surface current, and also lets you assign currents easily (G/num_coils basically).

The signs here are bc in my contour finding I assume the contours start at zeta=0 and the next contours are at positive zeta, but if G is negative for example, then contours increasing from 0 to +G actually corespond to negative zetas (since Phi ~ G*zeta)

Reply:
I want to make sure I understand this correctly:

Does this function always return coils with a coil at $(\theta=0,\phi=0)$? (For both helical and modular coils.) 

If so, that seems like an unnecessary restriction and we should at least document it clearly.

Reply:
I pick contours equally spaced in Phi and I have to start somewhere, so i choose to start at the point corresponding to theta=0 phi=0. I would need more checks to allow custom contour values to be passed in which I'd rather not do in this PR. I will note this in the docstring